src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -1476,6 +1506,48 @@ void SystemDictionaryShared::validate_before_archiving(InstanceKlass* k) {
   }
 }
 
+class UnregisteredClassesDuplicationChecker : StackObj {
+  GrowableArray<InstanceKlass*> _list;
+  Thread* _thread;
```

Comment:
seems _thread only used at line 1542, could remove the field and use Thread::current() in the site.

Reply:
_thread is used in the loop that iterates over all the classes in the _dumptime_table, so I wanted to avoid calling Thread::current() for each iteration.